compiler/optimizer/InductionVariable.cpp

```diff
@@ -4946,70 +4978,123 @@ TR::Node *TR_LoopStrider::signExtend(
                child->getOpCodeValue() == TR::lload,
                "exload l2i n%un's child should be an lload\n",
                node->getGlobalIndex());
-            ext = child;
+            result.extended = child;
+            result.cancelsExt = true;
+            result.cancelsTrunc = true;
             }
          break;
 
       case TR::iadd:
-         ext = signExtendBinOp(TR::ladd, node, exLoads, memo);
+         result = signExtendBinOp(TR::ladd, node, exLoads, memo);
          break;
 
       case TR::isub:
-         ext = signExtendBinOp(TR::lsub, node, exLoads, memo);
+         result = signExtendBinOp(TR::lsub, node, exLoads, memo);
          break;
 
       case TR::imul:
-         ext = signExtendBinOp(TR::lmul, node, exLoads, memo);
+         result = signExtendBinOp(TR::lmul, node, exLoads, memo);
          break;
 
       default:
-         break; // just don't set ext
+         break; // just don't populate result
       }
 
-   if (ext != NULL)
+   if (result.extended != NULL)
       {
-      memo.insert(std::make_pair(node->getGlobalIndex(), ext));
+      result.extended->incReferenceCount();
+      memo.insert(std::make_pair(node->getGlobalIndex(), result));
       if (trace())
          traceMsg(comp(),
             "[Sign-Extn] sign-extended n%un %s into n%un %s\n",
             node->getGlobalIndex(),
             node->getOpCode().getName(),
-            ext->getGlobalIndex(),
-            ext->getOpCode().getName());
+            result.extended->getGlobalIndex(),
+            result.extended->getOpCode().getName());
       }
 
-   return ext;
+   TR_ASSERT(
+      (result.extended != NULL) == (result.cancelsExt || result.cancelsTrunc),
+      "n%un %s should extend if and only if a conversion is canceled\n",
+      node->getGlobalIndex(),
+      node->getOpCode().getName());
+
+   return result;
    }
 
 // part of implementation of signExtend (above)
-TR::Node *TR_LoopStrider::signExtendBinOp(
+TR_LoopStrider::SignExtEntry TR_LoopStrider::signExtendBinOp(
    TR::ILOpCodes op64,
    TR::Node *node,
    TR::NodeChecklist &exLoads,
-   NodeMap &memo)
+   SignExtMemo &memo)
    {
+   const SignExtEntry rejected;
    if (!node->cannotOverflow())
-      return NULL;
-
-   TR::Node *extLeft = signExtend(node->getChild(0), exLoads, memo);
+      return rejected;
+
+   static const char * const moveDisableEnv =
+      feGetEnv("TR_disableLoopStriderMoveSignExtIntoChild");
+   static const bool moveDisable =
+      moveDisableEnv != NULL && moveDisableEnv[0] != '\0';
```

Comment:
Ok - we need to get this fix merged, but the usage is non-standard compared to the rest of the JIT where exporting a variable (regardless of value or non-value) is enough and you have to unset to clear it. We should probably try to standardize the usage across opts, but since there are other instances already we can merge this and look to clean-up the usage as another commit after this is sorted.